packages/flutter/lib/src/cupertino/segmented_control.dart

```diff
@@ -0,0 +1,588 @@
+    int index = 0;
+    for (T currentKey in widget.children.keys) {
+      final Key childKey = ValueKey<int>(index);
+      final TextStyle textStyle = DefaultTextStyle.of(context).style.copyWith(
+            color: (_selectedValue == childKey)
+                ? CupertinoColors.white
+                : CupertinoColors.activeBlue,
+          );
+      final IconThemeData iconTheme = IconThemeData(
+        color: (_selectedValue == childKey)
+            ? CupertinoColors.white
+            : CupertinoColors.activeBlue,
+      );
+
+      Widget child = widget.children[currentKey];
+      child = GestureDetector(
+        onTapDown: (TapDownDetails event) {
+          _onTapDown(childKey);
+        },
+        onTapUp: (TapUpDetails event) {
+          _onTapUp();
+        },
+        onTapCancel: () {
+          _onTapCancel();
+        },
+        onTap: () {
+          _onTap(currentKey, childKey);
+        },
+        child: new IconTheme(
+            data: iconTheme,
+            child: DefaultTextStyle(
+              style: textStyle,
+              overflow: TextOverflow.ellipsis,
+              child: Semantics(
+                  inMutuallyExclusiveGroup: true,
+                  selected: _selectedValue != null
+                      ? _selectedValue.value == index
+                      : false,
+                  child: child),
+            )),
+      );
+      _gestureChildren.add(child);
+      index += 1;
+    }
+
+    final Widget box = new _SegmentedControlRenderWidget(
+      children: _gestureChildren,
+      selectedValue: _selectedValue,
+      pressedValue: _pressedValue,
+    );
+
+    return Padding(
+      padding: _kHorizontalItemPadding.resolve(Directionality.of(context)),
+      child: UnconstrainedBox(
+        constrainedAxis: Axis.horizontal,
+        child: box,
+      ),
+    );
+  }
+}
+
+class _SegmentedControlRenderWidget extends MultiChildRenderObjectWidget {
+  _SegmentedControlRenderWidget({
+    Key key,
+    List<Widget> children = const <Widget>[],
+    @required this.selectedValue,
+    @required this.pressedValue,
+  }) : super(key:key, children: children);
+
+  final ValueKey<int> selectedValue;
+  final ValueKey<int> pressedValue;
+
+  @override
+  RenderObject createRenderObject(BuildContext context) {
+    return new _RenderSegmentedControl(
+      textDirection: Directionality.of(context),
+      selectedValue: selectedValue,
+      pressedValue: pressedValue,
+    );
+  }
+
+  @override
+  void updateRenderObject(
+      BuildContext context, _RenderSegmentedControl renderObject) {
+    renderObject.markNeedsPaint();
+    renderObject
+      ..textDirection = Directionality.of(context)
+      ..selectedValue = selectedValue
+      ..pressedValue = pressedValue;
+  }
+}
+
+class _SegmentedControlContainerBoxParentData extends ContainerBoxParentData<RenderBox> {
+}
+
+class _RenderSegmentedControl extends RenderBox with ContainerRenderObjectMixin<RenderBox, ContainerBoxParentData<RenderBox>>{
```

Comment:
This should be a `_RenderSegmentedControl<T>` so that you can get away from the `ValueKey<int>`s.